src/containers/flows/SetBuyNow/index.tsx

```diff
@@ -120,11 +132,22 @@ const SetBuyNowContainer = ({
 						library.getSigner(),
 					);
 				} else {
-					console.log('cancelling buy now');
 					tx = await sdk.zauction.cancelBuyNow(domainId, library.getSigner());
 				}
-				setCurrentStep(Step.SettingBuyNow);
-				await tx.wait();
+
+				// Handle transaction request
+				try {
+					setCurrentStep(Step.SettingBuyNow);
+					await tx?.wait();
+					// Timeout to prevent jolt
+					if (!amount) {
+						await new Promise((r) => setTimeout(r, 500));
+					}
+				} catch (e) {
+					setCurrentStep(Step.SetBuyNow);
+					setError(ERRORS.TRANSACTION);
```

Comment:
Same issue as earlier with continued execution after the `catch`. `return` or `throw` from `catch`.